substrate/client/src/client.rs

```diff
@@ -207,36 +209,29 @@ impl<B, E> Client<B, E> where
 		self.storage(id, &StorageKey(b":code".to_vec())).map(|data| data.0)
 	}
 
-	/// Clone a new instance of Executor.
-	pub fn clone_executor(&self) -> E where E: Clone {
-		self.executor.clone()
+	/// Get the set of authorities at a given block.
+	pub fn authorities_at(&self, id: &BlockId) -> error::Result<Vec<AuthorityId>> {
+		self.executor.call(id, "authorities",&[])
```

Comment:
what is your plan for having light clients memoize this and code changes? How will that fit into the abstraction?

Under the current consensus algorithm, we can expect certain runtimes to make a digest entry every time the authorities or code change for the next block. Under the planned consensus algorithm this change will be applied at some point beyond finality of the signal.

Reply:
Now: detect change in `import_block` method (digest is available there for light clients) => reset cache (not sure yet - where cache must be placed - in `OnDemand` service, or at the `LightBackend`/`LightCallExecutor`).

Then: not sure, but I guess the same strategy will work, but cache reset must be scheduled, instead of executing right now.

Reply:
OK - will add some meta-db + make cache backed by this db then.

Reply:
btw - the question from the future PRs :) do we need the same data on full nodes, or we're only considering archive nodes right now (=> all states are available => cache is not required)?

Reply:
right now let's consider only archive nodes